src/internal/pfsdb/commits_test.go

```diff
@@ -0,0 +1,585 @@
+				if prevCommit != nil {
+					commitInfo.ParentCommit = prevCommit.Commit
+				}
+				expectedInfos[i] = commitInfo
+				require.NoError(t, pfsdb.CreateCommit(ctx, tx, commitInfo))
+				createBranch(ctx, t, tx, commitInfo.Commit)
+				if i == size-1 { // the first commit will be missing branch information, so we need to add it.
+					require.NoError(t, pfsdb.UpdateCommit(ctx, tx, 1, commitInfo))
+				}
+				if prevCommit != nil {
+					require.NoError(t, pfsdb.PutCommitAncestryByCommitKeys(ctx, tx, prevCommit.Commit, commitInfo.Commit))
+					expectedInfos[i+1].ChildCommits = append(expectedInfos[i+1].ChildCommits, commitInfo.Commit)
+				}
+				prevCommit = commitInfo
+			}
+			iter, err := pfsdb.ListCommitTx(ctx, tx, nil, true)
+			require.NoError(t, err, "should be able to list repos")
+			checkOutput(ctx, t, iter, expectedInfos)
+			return nil
+		}))
+		iter, err := pfsdb.ListCommit(ctx, db, nil, true)
+		require.NoError(t, err, "should be able to list repos")
+		checkOutput(ctx, t, iter, expectedInfos)
+	})
+}
+
+func TestListCommitsFilter(t *testing.T) {
+	repos := []string{"a", "b", "c"}
+	size := 330
+	expectedInfos := make([]*pfs.CommitInfo, 0)
+	commitSetIds := make([]string, 0)
+	commits := make([]*pfs.CommitInfo, 0)
+	filter := pfsdb.CommitListFilter{
+		pfsdb.CommitRepos:    []string{"b"},
+		pfsdb.CommitOrigins:  []string{pfs.OriginKind_ORIGIN_KIND_UNKNOWN.String(), pfs.OriginKind_USER.String()},
+		pfsdb.CommitBranches: []string{"master"},
+		pfsdb.CommitProjects: []string{"default"},
+	}
+	testCommitDataModelAPI(t, func(ctx context.Context, t *testing.T, db *pachsql.DB) {
+		require.NoError(t, dbutil.WithTx(ctx, db, func(ctx context.Context, tx *pachsql.Tx) error {
+			for i := 0; i < size; i++ {
+				commitInfo := testCommit(ctx, t, tx, repos[i%len(repos)])
+				if commitInfo.Commit.Repo.Name == "b" && i%10 == 0 {
+					expectedInfos = append(expectedInfos, commitInfo)
+					commitSetIds = append(commitSetIds, commitInfo.Commit.Id)
+				}
+				commits = append(commits, commitInfo)
+			}
+			filter[pfsdb.CommitSetIDs] = commitSetIds
+			for _, commitInfo := range commits {
+				require.NoError(t, pfsdb.CreateCommit(ctx, tx, commitInfo))
+				createBranch(ctx, t, tx, commitInfo.Commit)
+			}
+			require.NoError(t, pfsdb.UpdateCommit(ctx, tx, 1, commits[0]))
+			iter, err := pfsdb.ListCommitTx(ctx, tx, filter, false)
+			require.NoError(t, err, "should be able to list repos")
+			checkOutput(ctx, t, iter, expectedInfos)
+			return nil
+		}))
+		iter, err := pfsdb.ListCommit(ctx, db, filter, false)
+		require.NoError(t, err, "should be able to list repos")
+		checkOutput(ctx, t, iter, expectedInfos)
+	})
+}
+
+type commitTestCase func(context.Context, *testing.T, *pachsql.DB)
+
+func testCommitDataModelAPI(t *testing.T, testCase commitTestCase) {
+	ctx := pctx.TestContext(t)
+	db, err := dbutil.NewDB(dockertestenv.NewTestDBOptions(t)...)
+	require.NoError(t, err)
+	t.Cleanup(func() {
```

Comment:
If you use `dockertestenv.NewTestDB(t)` then you don't need to cleanup here, since it does it for you automatically.

Reply:
good catch, I think I had this setup because I was explicitly creating a branches collection before.